File: android2_lesson03/src/java/java/com/example/ama/android2_lesson03/repo/data/base/MarkerManager.java
package com.example.ama.android2_lesson03.repo.data.base;

import com.example.ama.android2_lesson03.repo.model.SimpleMarker;

import java.util.ArrayList;

/**
 * Interface describes basic methods for managing favorites markers
 */
public interface MarkerManager {
    ArrayList<SimpleMarker> getAllMarkers();

    void addMarker(SimpleMarker marker);

    void updateMarker(SimpleMarker marker, String newName);

    void deleteMarker(SimpleMarker marker);
}

File: android2_lesson03/src/kotlin/java/com/example/ama/android2_lesson03/repo/data/base/MarkerManager.kt
package com.example.ama.android2_lesson03.repo.data.base

import com.example.ama.android2_lesson03.repo.data.model.SimpleMarker

/**
 * Interface describes basic methods for managing favorites markers
 */
interface MarkerManager {
    fun getAllMarkers(): ArrayList<SimpleMarker>
    fun addMarker(marker: SimpleMarker)
    fun deleteMarker(marker: SimpleMarker)
    fun updateMarker(marker: SimpleMarker, newName: String)
    fun isMarkerExists(marker: SimpleMarker): Boolean
}